pkg/util/policy/agent/agent.py

```diff
@@ -0,0 +1,447 @@
+        return
+
+    def do_POST(self):
+        """
+        Processes POST requests
+        extracts romana policy definition objects and passes it down for implementation
+    
+        Expected structure: { "method" : "ADDED|DELETED", "policy_definition" : "NP" }
+        """
+
+        self.send_header('Content-type','text/html')
+        self.end_headers()
+        # Send the html message
+        headers = Message(StringIO(self.headers))
+        raw_data = self.rfile.read(int(headers["Content-Length"]))
+        try:
+            json_data = simplejson.loads(raw_data)
+        except Exception, e:
+            logging.warning("Cannot parse %s" % raw_data)
+            return
+
+        # TODO This endpoint shoud really go away, only used for prototyping
+        # while new policy fromat being discussed.
+        if self.path=="/ns-update":
+            logging.info("Request hit /ns-update")
+
+            tid = json_data.get('tid')
+            logging.info("tid is %s" % tid)
+
+            isolated = json_data.get('isolated')
+            logging.info("isolated is %s" % isolated)
+
+            iptables_rules = get_current_iptables()
+            TENANT_POLICY_VECTOR_CHAIN = "ROMANA-T%s" % tid
+            logging.info("TENANT_POLICY_VECTOR_CHAIN is %s" % TENANT_POLICY_VECTOR_CHAIN)
+
+            
+            # Parse out a list of chain names out of current rules,
+            # use it to avoid duplication when adding new chains.
+            existing_chains = [ k.split(" ")[0][1:] for k in iptables_rules if k.startswith(":") ]
+
+            tenant_vector_chain_exists = TENANT_POLICY_VECTOR_CHAIN in existing_chains
+            logging.info("tenant_vector_chain_exists is %s" % tenant_vector_chain_exists)
+
+            ALLOW_ANY_VECTOR = "-A %s -j ACCEPT" % TENANT_POLICY_VECTOR_CHAIN
+            logging.info("ALLOW_ANY_VECTOR is %s" % ALLOW_ANY_VECTOR)
+
+            allow_any_vector_exists = ALLOW_ANY_VECTOR in iptables_rules
+            logging.info("allow_any_vector_exists is %s" % allow_any_vector_exists)
+
+            filter_idx = iptables_rules.index('*filter')
+
+            if not tenant_vector_chain_exists:
+                logging.info("Tenant policy vector chain does not exist - creating")
+                iptables_rules.insert(filter_idx + 1, ":%s - [0:0]" % TENANT_POLICY_VECTOR_CHAIN)
+
+            if allow_any_vector_exists:
+                if isolated:
+                    logging.info("Enabling isolation")
+                    iptables_rules.remove(ALLOW_ANY_VECTOR)
+            else:
+                if not isolated:
+                    logging.info("Disabling isolation")
+                    iptables_rules.insert(filter_rules_idx(iptables_rules), ALLOW_ANY_VECTOR)
+            apply_new_ruleset(iptables_rules)
+            return
+
+
+        method = json_data.get('method')
+        policy_def = json_data.get('policy_definition')
+
+        if method not in [ 'ADDED', 'DELETED' ] or not policy_def:
```

Comment:
This whole "method in the request body" needs to change. We should use the HTTP verbs POST, PUT and DELETE for adding, updating and deleting. That shouldn't all be carried inside of a POST request.

We can leave this for now, since this is temporary code, but please put a big TODO on there.